tap_core/src/rav.rs

```diff
@@ -106,3 +119,34 @@ impl ReceiptAggregateVoucher {
         })
     }
 }
+
+impl Aggregate<SignedReceipt> for ReceiptAggregateVoucher {
+    fn aggregate_receipts(
+        receipts: &[ReceiptWithState<Checked, SignedReceipt>],
+        previous_rav: Option<EIP712SignedMessage<Self>>,
+    ) -> Result<Self, Error> {
+        if receipts.is_empty() {
+            return Err(Error::NoValidReceiptsForRavRequest);
+        }
+        let allocation_id = receipts[0].signed_receipt().message.allocation_id;
+        let receipts = receipts
+            .iter()
+            .map(|rx_receipt| rx_receipt.signed_receipt().clone())
+            .collect::<Vec<_>>();
+        ReceiptAggregateVoucher::aggregate_receipts(
+            allocation_id,
+            receipts.as_slice(),
+            previous_rav,
+        )
+    }
+}
+
+impl WithValueAndTimestamp for ReceiptAggregateVoucher {
+    fn value(&self) -> u128 {
+        self.valueAggregate
+    }
+
+    fn timestamp_ns(&self) -> u64 {
+        self.timestampNs
```

Comment:
```suggestion
        self.timestampNs
```

Interesting casing!

Reply:
This is a weird one hahaha, since we are using SolStruct, if we don't pass it camelCase, the EIP712 Header becomes different than in Solidity.

ReceiptAggregateVoucher is the only thing sent to the blockchain and must match Solidity code. At the time I couldn't find any alloy annotation to convert it to camelCase automatically.

Reply:
```rust
#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
```

No good?

Reply:
alloy doesn't use Serde to generate the header, it has it's own macros